### PR TITLE
[SPARK-8988][YARN] Make sure driver log links appear in secure cluste…

### PR DESCRIPTION
**Pull request by harishreedharan**
…r mode.

The NodeReports API currently used does not work in secure mode since we do not get RM tokens. Instead this patch just uses environment vars exported by YARN to create the log links.

### CODE REVIEW

File: yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala [hunk 1]
```diff
@@ -19,6 +19,8 @@ package org.apache.spark.scheduler.cluster
 
 import java.net.NetworkInterface
 
+import org.apache.hadoop.yarn.api.ApplicationConstants.Environment
```
Comment (vanzin) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala]: nit: ordering

File: yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala [hunk 2]
```diff
@@ -64,68 +66,29 @@ private[spark] class YarnClusterSchedulerBackend(
...
+      )
+      val user = Utils.getCurrentUserName()
+      val httpScheme = if (yarnHttpPolicy == "HTTPS_ONLY") "https://" else "http://"
+      val baseUrl = s"$httpScheme$httpAddress/node/containerlogs/$containerId/$user"
+      logDebug(s"Base URL for logs: $baseUrl")
+      driverLogs = Some(Map(
+        "stderr" -> s"$baseUrl/stderr?start=-4096",
+        "stdout" -> s"$baseUrl/stdout?start=-4096"))
     } catch {
       case e: Exception =>
```
Comment (vanzin) [inline yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerBackend.scala]: It doesn't seem like there's any exception that can be thrown anymore.

**Comment by vanzin** [comment]
LGTM if it works. Are these env variables available to containers even if they're not listed in `yarn.nodemanager.env-whitelist`?

**Comment by harishreedharan** [comment]
@vanzin I am not sure about that. I can check with some YARN folks or perhaps @tgravescs knows.

**Comment by tgravescs** [comment]
Yes the variables are always there.  They don't need to be in the whitelist.

**Comment by vanzin** [comment]
I verified that too (those variables are not in my whitelist) by running this pyspark code:

    map(lambda x: print repr(x), sc.parallelize(range(0, 1)).map(lambda x: os.environ).collect())

LGTM. I'll let it sit a bit to see if anyone else has comments.

**Comment by harishreedharan** [comment]
I verified on a secure cluster - looks like the links appear now.

**Comment by tgravescs** [comment]
looks good.

**Comment by vanzin** [comment]
Merged to master. thanks!

**Pull request closed without merging**